### PR TITLE
📖 Document Kubernetes bump

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
First try to add some docs around this. Let's try to incrementally improve

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sbueringer** [comment]
/assign @fabriziopandini @chrischdi 

(already updated the issue)

**Review by fabriziopandini — COMMENTED** [review]
/lgtm 
That's a great start, we can iterate in future e.g by adding reference PRs

File: packaging/flavorgen/cloudprovider/cpi/cpi.yaml [hunk 1]
```diff
@@ -155,7 +155,7 @@ spec:
       priorityClassName: system-node-critical
       containers:
       - name: vsphere-cpi
-        image: gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.30.0-rc.0
+        image: gcr.io/cloud-provider-vsphere/cpi/release/manager:vx.y.z
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply packaging/flavorgen/cloudprovider/cpi/cpi.yaml:158]: Ah good point. I'll revert. Didn't remember

[the remark this answers is not in the document]
Comment (sbueringer) [reply packaging/flavorgen/cloudprovider/cpi/cpi.yaml:158]: I was just grepping  a bit to check if I missed anything. Then first bumped it manually just to realize that it doesn't matter :D

File: .github/ISSUE_TEMPLATE/release_tracking.md [hunk 2]
```diff
@@ -15,8 +15,23 @@ Early in the cycle:
...
+    * Build new OVA images via image-builder
+    * Make them available in the CI environment
+    * Publish them via a GitHub release (e.g. [templates/v1.30.0](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/releases/tag/templates/v1.30.0))
+    * Update `README.md` accordingly
+  * [ ] Bump e2e tests
+    * Add the new image to `internal/test/helpers/vcsim/model.go`
+    * Bump template env variables in `test/e2e/config/vsphere.yaml` and `test/e2e/config/config-overrides-example.yaml`
+      * Also bump `KUBERNETES_VERSION_*`
+      * Also bump `KUBERNETES_VERSION_LATEST_CI` to the next minor version
+      * Also bump `CPI_IMAGE_K8S_VERSION`
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply .github/ISSUE_TEMPLATE/release_tracking.md:30]: Thx!

**Comment by sbueringer** [comment]
Fixed & updated the issue, PTAL

**Comment by sbueringer** [comment]
Yup let's go ahead and merge

/hold cancel

**Pull request merged**